### PR TITLE
Add a page for WebHID API

### PR DESCRIPTION
**Pull request by SnorlaxYum**
Add a page for WebHID API. Later I will add pull requests one by one for the global interfaces `HID`, `HIDConnectionEvent`, `HIDInputReportEvent`, `HIDDevice`.

As for compat data, already added a pull request: https://github.com/mdn/browser-compat-data/pull/9910

And it seems the backend didn't recognize the correct specification url, it should be: https://wicg.github.io/webhid/ . So maybe ur backend developers should work to fix that recognization problem?

**To-do list about associated pages:** (I think maybe I'll do them one by one in future PRs, it's not recommended to make a substantial PR)
- [x] `navigator.hid` - https://github.com/mdn/content/pull/4345
- [ ] interface pages
  - [x] `HID` - https://github.com/mdn/content/pull/4368
  - [ ] `HIDConnectionEvent`
  - [ ] `HIDInputReportEvent`
  - [ ] `HIDConnectionEvent`

### CODE REVIEW

File: files/en-us/web/api/webhid_api/index.html [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    <dt>{{domxref("HID")}}</dt>
+    <dd>The global HID interface.</dd>
+</dl>
+<dl>
+    <dt>{{domxref("HIDDevice")}}</dt>
+    <dd>Filter the device. Valid if filter.productId, filter.vendorId, filter.invalid and filter.usage exist.</dd>
+</dl>
+<dl>
+    <dt>{{domxref("HIDInputReportEvent")}}</dt>
+    <dd>The Event about HID Input Report.</dd>
```
Comment (jpmedley) [inline files/en-us/web/api/webhid_api/index.html:26]:
```suggestion
    <dd>Passed to {{domxref("HIDDevice. oninputreport")}} when a user interacts with the specified device.</dd>
```

File: files/en-us/web/api/webhid_api/index.html [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+</dl>
+<dl>
+    <dt>{{domxref("HIDConnectionEvent")}}</dt>
+    <dd>The Event about HID Connection.</dd>
```
Comment (jpmedley) [inline files/en-us/web/api/webhid_api/index.html:30]:
```suggestion
    <dd>Passed to {{domxref("HID.onconnect")}} and {{domxref("HID.ondisconnect")}} when a device is connected or disconnected.</dd>
```

File: files/en-us/web/api/webhid_api/index.html [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+</dl>
+
+<h2 id="Examples">Examples</h2>
+
+<p>Retrieve devices and log the device names to the console.</p>
```
Comment (jpmedley) [inline files/en-us/web/api/webhid_api/index.html:35]: Please use complete sentences when introducing examples.

File: files/en-us/web/api/webhid_api/index.html [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    });
+});</pre>
+
+<p>Register event listeners for connection and disconnection of HID devices.</p>
+
+<pre class="brush: js">navigator.hid.addEventListener('connect', ({device}) => {
+    console.log(`HID connected: ${device.productName}`);
+    });
+
+    navigator.hid.addEventListener('disconnect', ({device}) => {
```
Comment (jpmedley) [inline files/en-us/web/api/webhid_api/index.html:50]:
```suggestion
    navigator.hid.addEventListener('disconnect', (device) => {
```

**Review by jpmedley — CHANGES_REQUESTED** [review]
Hi,

Thank you for your submission. The first paragraph shows that you can write. The feedback I have is to make this fit better with how we do things in MDN. (Any website that depends on writing will likely have requirements specific to it.) In case you haven't found them yet, [here's MDN's guidelines](https://developer.mozilla.org/en-US/docs/MDN/Contribute/Howto/Write_an_API_reference).

One general comment I have is that descriptions of things need to be very specific. I've made some suggestions. 

Let me know if you have any questions. I'm here to help.

Joe

File: files/en-us/web/api/webhid_api/index.html [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+slug: Web/API/WebHID_API
+tags:
+  - API
+  - Advanced
+  - WebHID
+  - WebHID API
+---
+<div>{{DefaultAPISidebar("WebHID API")}}</div>
+
+<p>A HID (Human Interface Device) is a type of device that takes input from or provides output to humans. It also refers to the HID protocol, a standard for bi-directional communication between a host and a device that is designed to simplify the installation procedure. The HID protocol was originally developed for USB devices but has since been implemented over many other protocols, including Bluetooth.</p>
```
[the remark this answers is not in the document]
Comment (Ryuno-Ki) [reply files/en-us/web/api/webhid_api/index.html:12]: Is there a legal public version of it?
I've heard of Chicago Manual of Style from A List Apart the first time. A research brings up a book that's still for sale.
Kinda bummer, that we refer to a styleguide, which is not publicly available for everyone to check :-/

File: files/en-us/web/api/webhid_api/index.html (hunk 4 above)
Comment (SnorlaxYum) [reply files/en-us/web/api/webhid_api/index.html:50]: The [original example](https://wicg.github.io/webhid/#example-1) was doing the destructuring assignment

File: files/en-us/web/api/webhid_api/index.html (hunk 5 above)
[the remark this answers is not in the document]
Comment (jpmedley) [reply files/en-us/web/api/webhid_api/index.html]: It's a rather pricy book. I don't generally hold people to standards in books they don't own. This is common enough that was worth sharing that its a thing.

File: files/en-us/web/api/webhid_api/index.html [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+---
+<div>{{DefaultAPISidebar("WebHID API")}}</div>
+
+<p>A Human Interface Device (HID) is a type of device that takes input from or provides output to humans. It also refers to the HID protocol, a standard for bi-directional communication between a host and a device that is designed to simplify the installation procedure. The HID protocol was originally developed for USB devices but has since been implemented over many other protocols, including Bluetooth.</p>
+
+<h2 id="Interfaces">Interfaces</h2>
+
+<dl>
+    <dt>{{domxref("HID")}}</dt>
+    <dd>provides methods for connecting to HID devices, listing attached HID devices and event handlers for connected HID devices.</dd>
```
Comment (jpmedley) [inline files/en-us/web/api/webhid_api/index.html:18]:
```suggestion
    <dd>Provides methods for connecting to HID devices, listing attached HID devices and event handlers for connected HID devices.</dd>
```

File: files/en-us/web/api/webhid_api/index.html [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+</dl>
+<dl>
+    <dt>{{domxref("HIDDevice")}}</dt>
+    <dd>represents an HID device. It's possible for a single physical device to be represented by multiple `HIDDevice` obects.</dd>
```
Comment (jpmedley) [inline files/en-us/web/api/webhid_api/index.html:22]:
```suggestion
    <dd>Represents an HID device. It's possible for a single physical device to be represented by multiple `HIDDevice` obects.</dd>
```

File: files/en-us/web/api/webhid_api/index.html [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+</dl>
+<dl>
+    <dt>{{domxref("HIDInputReportEvent")}}</dt>
+    <dd>passed to {{domxref("HIDDevice.oninputreport")}} when an input report is received from the any associated HID device.</dd>
```
Comment (jpmedley) [inline files/en-us/web/api/webhid_api/index.html:26]:
```suggestion
    <dd>Passed to {{domxref("HIDDevice.oninputreport")}} when an input report is received from the any associated HID device.</dd>
```

File: files/en-us/web/api/webhid_api/index.html [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+</dl>
+<dl>
+    <dt>{{domxref("HIDConnectionEvent")}}</dt>
+    <dd>passed to {{domxref("HID.onconnect")}} and {{domxref("HID.ondisconnect")}} when a device is connected or disconnected.</dd>
```
Comment (jpmedley) [inline files/en-us/web/api/webhid_api/index.html:30]:
```suggestion
    <dd>Passed to {{domxref("HID.onconnect")}} and {{domxref("HID.ondisconnect")}} when a device is connected or disconnected.</dd>
```

**Review by jpmedley — CHANGES_REQUESTED** [review]
I'm making these requests with an apology. I incorrectly capitalized some of my suggestions. I don't like to subject volunteers to multiple rounds of review. Please forgive and accept the changes.

**Comment by rachelandrew** [comment]
@SnorlaxYum Specs need to be added to Yari if they don't already exist in the system (which is likely the case if you are documenting something with no existing interfaces documented already). 

I've added this one in https://github.com/mdn/yari/pull/3599 however even once merged it may take a little while to show up. Add the spec tables as you have been doing and it will resolve once it is all there.

**Comment by SnorlaxYum** [comment]
@rachelandrew awesome, thanks.

File: files/en-us/web/api/webhid_api/index.html [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+    <dd>Passed to {{domxref("HIDDevice.oninputreport")}} when an input report is received from any associated HID device.</dd>
+</dl>
+<dl>
+    <dt>{{domxref("HIDConnectionEvent")}}</dt>
+    <dd>Passed to {{domxref("HID.onconnect")}} and {{domxref("HID.ondisconnect")}} when a device is connected or disconnected.</dd>
+</dl>
+
+<h2 id="Examples">Examples</h2>
+
+<p>You can connect to a device with {{domxref("HID.requestDevices")}}. In this case, we gonna select from all the available devices.</p>
```
Comment (rachelandrew) [inline files/en-us/web/api/webhid_api/index.html:35]:
```suggestion
<p>You can connect to a device with {{domxref("HID.requestDevices")}}. In this case, we select from all the available devices.</p>
```

**Review by jpmedley — APPROVED** [review]

**Comment by jpmedley** [comment]
Thank you for the help.

**Pull request merged**